### ISSUE TITLE
stylo: Align restyle hints meaning between Servo and Gecko

### ISSUE THREAD

**Issue by emilio** [servo/servo#12710]
TL;DR:
- `RESTYLE_SELF` should not (always) run selector matching on descendants (that's `eRestyle_Subtree`, which currently is lacking on Servo). This is a known (hopefully soon-to-be-addressed issue). We treat both like the same right now.
- `RESTYLE_DESCENDANTS` and `eRestyle_SomeDescendants` share binary representations right now, but the meanings for Gecko and Servo are different (all vs. some). In practice this doesn't matter much since it's impossible for a Gecko-generated `eRestyle_SomeDescendants` to reach the Servo restyle manager, but we should be aware of this and presumably give it a more meaningful name, or pick another spot, or something.

cc @heycam @bholley

**Comment by bholley** [comment]
@emilio Per previous discussion I think we should track stylo on bugzilla. Can you file a corresponding bug and point to this issue?

**Comment by emilio** [comment]
Sure, filled https://bugzilla.mozilla.org/show_bug.cgi?id=1291786

**Comment by nox** [comment]
The bugzilla ticket was closed, so I assume this was done.
